### DISCUSSION TITLE
Pixel aspect ratio

### DISCUSSION THREAD

**Discussion by SteffenDuenner** [ggarra13/mrv2#21] [Ideas]
Hi there,

it would be awesome if mrv2 supported pixel aspect ratio for viewing anamorphic footage (which I currently have to deal with a lot ;) ).

Thanks

P.S.: Is there something like a public road map for the development?

**Comment by ggarra13** [comment]
mrv2 reads the pixel aspect ratio of the image (OpenEXR and DPX) and from movies' metadata.  It should stretch the image in proportion to that.
If you have a sample image or movie file that mrv2 does not do it, please be so kind to share it (privately if needs be) as it would be a bug or missing feature.

**Comment by SteffenDuenner** [comment]
Actually you're right!

The last time I reviewed the footage the aspect ratio was off. And additionally the stuff coming back from color grading didn't contain any metadata so mrv2 showed me skewed stuff. Now all is fine. :)
Sorry for the noise.

**Reply by SteffenDuenner** [reply]
By the way, if the footage is missing metadata it would still be awesome if we could manually set an aspect ratio, just like we can override the frame rate.

**Reply by SteffenDuenner** [reply]
Currently I have to deal with anamorphic footage that got its metadata garbled during grading.
So mrv2 displays all the clips with the wrong aspect. It would be helpful, if we could manually override the pixel aspect ratio just like the playback frame rate:
![image](https://user-images.githubusercontent.com/33224667/234792347-cb4ce1df-6804-4e5d-b099-e1c94dad8ea1.png)

**Comment by SteffenDuenner** [comment]
Sorry to revive this old thread, but I again have to deal with anamorphic footage (PAR 2:1) that isn't properly recognized by mrv2.
VLC, Celluloid (MPV frontend) and Fusion do properly read the pixel aspect ratio and display the correct video size while mrv2 displays square pixels.

I attached a ProRes 422 Proxy file to show the problem.
[PAR2_1_v01.mov.zip](https://github.com/user-attachments/files/18520339/PAR2_1_v01.mov.zip)

The perfect solution would be a) mrv2 recognizes the correct pixel aspect ratio from the file and b) mrv2 offers us to manually override the PAR in case we want to.

EDIT: I noticed that Github doesn't offer to view or download the ProRes file so I zipped it and attached it here.

**Comment by ggarra13** [comment]
Hi, Steffen.  For some reson I did not get this comment in my email box (maybe because it was too old).  I've fixed mrv2 to recognize pixel aspects properly on movie files and added the ability to set the pixel aspect, albeit it is not interactive (the movie has to be flushed and reloaded).  There's still no way to set the pixel aspect for all the other image formats (OpenEXR, DPX, etc).

**Reply by SteffenDuenner** [reply]
Thanks you very much. This already helps me a lot while working with anamorphic footage.

**Reply by SteffenDuenner** [reply]
Just installed the latest beta and the pixel aspect ratio update / fix is working nicely. Thanks :)
